### PR TITLE
Echo 92 royalties improvements

### PR DESCRIPTION
**Pull request by f-pto**

### CODE REVIEW

File: tzkt-client/src/main/kotlin/com/rarible/tzkt/royalties/RoyaltiesHandler.kt [hunk 1]
```diff
@@ -129,12 +130,31 @@ class RoyaltiesHandler(val bigMapKeyClient: BigMapKeyClient, val ipfsClient: IPF
...
+            val royaltiesKey = bigMapKeyClient.bigMapKeyWithName(royaltiesConfig.fxhashV2, "token_data", tokenId)
+            royaltiesMap = royaltiesKey.value as LinkedHashMap<String, String>
+            val royaltiesAmount = royaltiesMap["royalties"]!!.toLong()
+            val percentages = royaltiesMap["royalties_split"] as ArrayList<LinkedHashMap<String, String>>
+            percentages.forEach {
+                parts.add(Part(it["address"]!!, royaltiesAmount * it["pct"]!!.toLong() / 100))
+            }
+            //parts.add(Part(authorMap["author"]!!, royaltiesMap["royalties"]!!.toLong() * 10))
+        } catch (e: Exception) {
+            logger.warn("Could not parse royalties for token $tokenId with FXHASH pattern: ${e.message}")
```
Comment (mystdeim) [inline tzkt-client/src/main/kotlin/com/rarible/tzkt/royalties/RoyaltiesHandler.kt:149]: It's good to have token address as well for simplier debug in future. `token:tokenId`

File: tzkt-client/src/main/kotlin/com/rarible/tzkt/royalties/RoyaltiesHandler.kt [hunk 2]
```diff
@@ -198,4 +218,39 @@ class RoyaltiesHandler(val bigMapKeyClient: BigMapKeyClient, val ipfsClient: IPF
...
+
+    private suspend fun getRoyaltiesFromRoyaltiesManager(contract: String, tokenId: String): List<Part> {
+        val parts = mutableListOf<Part>()
+        var data: List<LinkedHashMap<String, String>>? = null
+        try {
+            val tokenKeyValue = "{\"address\":\"$contract\",\"nat\":$tokenId}"
+            val key = bigMapKeyClient.bigMapKeyWithName(royaltiesConfig.royaltiesManager, "royalties", URLEncoder.encode(tokenKeyValue, StandardCharsets.UTF_8.toString()))
+            data = key.value as List<LinkedHashMap<String, String>>
+        } catch (e: Exception) {
+            logger.warn("Could not parse royalties for token $tokenId with Royalties Manager (with token id) pattern: ${e.message}")
```
Comment (mystdeim) [inline tzkt-client/src/main/kotlin/com/rarible/tzkt/royalties/RoyaltiesHandler.kt:232]: It's good to have token address as well for simplier debug in future. Please add `token:tokenId` here and for all cases below

File: tzkt-client/src/test/kotlin/com/rarible/tzkt/royalties/RoyaltiesTests.kt [hunk 3]
```diff
@@ -248,6 +307,81 @@ class RoyaltiesTests : BaseClientTests() {
         )
     }
 
+    @Test
+    fun `should correctly fetch and parse Royalties Manager royalties for a whole collection`() = runBlocking<Unit> {
+        mock404()
+        mock404()
+        mock404()
```
Comment (mystdeim) [inline tzkt-client/src/test/kotlin/com/rarible/tzkt/royalties/RoyaltiesTests.kt:315]: Why do we need mock404 here 4 times?

File: tzkt-client/src/main/kotlin/com/rarible/tzkt/royalties/RoyaltiesHandler.kt [hunk 4]
```diff
@@ -53,7 +51,7 @@ class RoyaltiesHandler(val bigMapKeyClient: BigMapKeyClient, val ipfsClient: IPF
         //check rarible pattern in generated contract storage
         part = getRaribleRoyalties(contract, tokenId)
 
-        if (!part.isNullOrEmpty()) {
+        if (part.isNotEmpty()) {
             logger.info("Token $id royalties pattern is RARIBLE")
```
Comment (mystdeim) [inline tzkt-client/src/main/kotlin/com/rarible/tzkt/royalties/RoyaltiesHandler.kt]: I think it's better to have `token:tokenId` everywhere for simplification debug in future

File: tzkt-client/src/main/kotlin/com/rarible/tzkt/client/IPFSClient.kt [hunk 5]
```diff
@@ -3,14 +3,20 @@ package com.rarible.tzkt.client
...
 ) : BaseClient(webClient) {
-    suspend fun data(hash: String): JsonNode {
+    suspend fun ipfsData(hash: String): JsonNode {
         val content = invoke<String> {
             it.path(hash)
         }
         return ObjectMapper().readTree(content)
     }
+
+    suspend fun data(url: String): JsonNode {
```
Comment (mystdeim) [inline tzkt-client/src/main/kotlin/com/rarible/tzkt/client/IPFSClient.kt:19]: Maybe it would be simplier to deserialize Json String directly to Map here.

**Review by mystdeim — CHANGES_REQUESTED** [review]

File: tzkt-client/src/main/kotlin/com/rarible/tzkt/client/IPFSClient.kt (hunk 5 above)
Comment (f-pto) [reply tzkt-client/src/main/kotlin/com/rarible/tzkt/client/IPFSClient.kt:19]: The problem is that we can't really do it in a standard way as we don't know what to expect

**Review by mystdeim — APPROVED** [review]

**Pull request merged**